src/Model/Union.php

```diff
@@ -0,0 +1,464 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\data\FieldSqlExpression;
+use atk4\data\Model;
+use atk4\dsql\Expression;
+use atk4\dsql\Query;
+
+/**
+ * Union model combines multiple nested models through a UNION in order to retrieve
+ * it's value set. The beauty of this class is that it will add fields transparently
+ * and will map them appropriately from the nested model if you request
+ * those fields from the union model.
+ *
+ * For example if you are asking sum(amount), there is no need to fetch any extra
+ * fields from sub-models.
+ */
+class Union extends Model
+{
+    /** @const string */
+    public const HOOK_AFTER_SELECT = self::class . '@afterSelect';
+
+    /** @deprecated use HOOK_AFTER_SELECT instead - will be removed dec-2020 */
+    public const HOOK_AFTER_UNION_SELECT = self::HOOK_AFTER_SELECT;
+
+    /**
+     * Union model should always be read-only.
+     *
+     * @var bool
+     */
+    public $read_only = true;
+
+    /**
+     * Union normally does not have ID field. Setting this to null will
+     * disable various per-id operations, such as load().
+     *
+     * If you can define unique ID field, you can specify it inside your
+     * union model.
+     *
+     * @var string
+     */
+    public $id_field;
+
+    /**
+     * Contain array of array containing model and mappings.
+     *
+     * $union = [ [ $model1, ['amount'=>'total_gross'] ] , [$model2, []] ];
+     *
+     * @var array
+     */
+    public $union = [];
+
+    /**
+     * When aggregation happens, this field will contain list of fields
+     * we use in groupBy. Multiple fields can be in the array. All
+     * the remaining fields will be hidden (marked as system()) and
+     * have their "aggregates" added into the selectQuery (if possible).
+     *
+     * @var array|string
+     */
+    public $group;
+
+    /**
+     * When grouping, the functions will be applied as per aggregate
+     * fields, e.g. 'balance'=>['sum', 'amount'].
+     *
+     * You can also use Expression instead of array.
+     *
+     * @var array
+     */
+    public $aggregate = [];
+
+    /** @var string Derived table alias */
+    public $table = 'derivedTable';
```

Comment:
why that default?

Reply:
Table name alias is mandatory for such union model constructs.

Like
```
select *
from (
    select * from A
    union all
    select * from B
) derivedTable <-- this is mandatory and should be something quite unique, but not random (should be static to not mess with DB engine cache)
```

Reply:
makes sense